hyperopt/suggest.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from future import standard_library
+from builtins import object
+import logging
+import os
+
+import numpy as np
+
+from .utils import coarse_utcnow
+from . import base
+from . import tpe
+from . import hp
+
+from pandas import DataFrame
+from datetime import datetime
+
+
+class SuggestObject(object):
+    """Object for conducting search experiments.
+    """
+
+    def __init__(self, algo, domain, trials, rstate):
+        self.algo = algo
+        self.domain = domain
+        self.trials = trials
+        self.rstate = rstate
+
+    def serial_evaluate(self):
+        for trial in self.trials._dynamic_trials:
```

Comment:
What are the semantics of `trials` and `_dynamic_trials`?

Reply:
_dynamic_trials is the real list of experiments. 
trials are _dynamic_trials + other metadata and auxiliary functions (for example: argmin)